File: tests/cases/find_factors.py
# From: https://book.pythontips.com/en/latest/for_-_else.html
# Modified example from the official Python documentation
def find_factors(n):
    for i in range(2, n):
        for j in range(2, i):
            if i % j == 0:
                print(i, 'equals', j, '*', i/j)
                break
        else:
            # loop fell through without finding a factor
            print(i, 'is a prime number')

if __name__ == "__main__":
    find_factors(100)

File: tests/expected/find_factors.jl
function find_factors(n)
    for i = 2:n-1
        has_break = false
        for j = 2:i-1
            if (i % j) == 0
                println("$(i) equals $(j) * $(i / j)")
                has_break = true
                break
            end
        end
        if has_break != true
            println("$(i) is a prime number")
        end
    end
end

if abspath(PROGRAM_FILE) == @__FILE__
    find_factors(100)
end
